tests/cupy_tests/indexing_tests/test_generate.py

```diff
@@ -24,3 +24,95 @@ def test_ix_empty_ndarray(self, xp):
     @testing.numpy_cupy_array_list_equal()
     def test_ix_bool_ndarray(self, xp):
         return xp.ix_(xp.array([True, False] * 2))
+
+
+@testing.gpu
+class TestR_(unittest.TestCase):
+
+    _multiprocess_can_split_ = True
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_r_1(self, xp, dtype):
+        a = testing.shaped_arange((3, 4), xp, dtype)
+        b = testing.shaped_reverse_arange((2, 4), xp, dtype)
+        return xp.r_[a, b]
+
+    def test_r_8(self, xp, dtype):
+        a = testing.shaped_arange((3, 4), xp, dtype)
+        b = testing.shaped_reverse_arange((2, 4), xp, dtype)
+        c = testing.shaped_reverse_arange((1, 4), xp, dtype)
+        return xp.r_[a, b, c]
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_array_equal()
+    def test_r_2(self, xp, dtype):
+        a = testing.shaped_arange((1, 3), xp, dtype)
+        b = testing.shaped_arange((1, 3), xp, dtype)
+        return xp.r_[a, 0, 0, b]
+
+    def test_r_3(self, xp):
```

Comment:
`xp` is given only when functions are decorated with [NumPy-CuPy consistency check decorator](
http://docs.chainer.org/en/stable/cupy-reference/testing.html#numpy-cupy-consistency-check)  e.g 
`@testing.numpy_cupy_array_equal` etc..